pkg/streamingpromql/config.go

```diff
@@ -5,32 +5,35 @@ package streamingpromql
 import (
 	"flag"
 
+	"github.com/grafana/dskit/flagext"
 	"github.com/prometheus/prometheus/promql"
 )
 
 type EngineOpts struct {
-	CommonOpts     promql.EngineOpts
-	FeatureToggles FeatureToggles
+	CommonOpts promql.EngineOpts
+	MQEOpts    MQEOpts
```

Comment:
How about we call this something like `Features`?

I was imagining that most MQE-specific config would go on the `EngineOpts` struct directly, like `Pedantic` below, and we keep feature flags in their own struct.